### PR TITLE
feat(pruatcher) adding way to avoid check with labels

### PR DESCRIPTION
**Pull request by gkope**
In platform we have a few PRs that are not going to be merged today, but we all know that they exists and will be merged in a near future.

It's very bad getting notified about the same PRs everyday and making everyone ignore the poor PRUacher notifications, since we all know that a PR with a `do-not-merge` label can be ignored.

Just add the 'do-not-merge' label in your PR and it will be automatically ignored by PRUatcher.

image

@michelazzo don't know if you agree that this is a needed future or if it's better to stop avoiding let PRs open for so long even when they have a 'do-no-merge' label.

### CODE REVIEW

File: pruatcher.py [hunk 1]
```diff
@@ -80,6 +80,13 @@ def process_squads(squad, githubConf):
 
         pulls = github_response.json()
 
+        for pull in pulls:
```
Comment (shikow) [inline pruatcher.py]: Using many fors its not a pythonic way. You can use list comprehensions to make more pythonic and simple(http://www.pythonforbeginners.com/basics/list-comprehensions-in-python). One example: `pulls = [pull for pull in pulls if valid_pull(pull)]`

**Comment by gkope** [comment]
@shikow I made the requested modifications, also, I change it so we can get the label from a list in a variable on Lambda, not a fixed variable anymore

I'm also thinking about migrating this guy to K8S and making the list of repositories to be monitored more dynamic and with ci/cd.

File: pruatcher.py [hunk 2]
```diff
@@ -12,6 +12,8 @@
 logger.setLevel(logging.INFO)
 test_env = os.environ.get('test_env')
 
+whitelist_pr_label_list = [label.strip() for label in os.environ['WHITELIST_PR_LABEL'].split(',')]
```
Comment (shikow) [inline pruatcher.py]: :heart:

File: pruatcher.py [hunk 3]
```diff
@@ -80,7 +91,10 @@ def process_squads(squad, githubConf):
 
         pulls = github_response.json()
 
-        for pull in pulls:
+        ## Filtering only valid PRs
+        valid_pulls = [pull for pull in pulls if valid_pull(pull)]
```
Comment (shikow) [inline pruatcher.py]: :heart:

File: pruatcher.py (hunk 2 above)
Comment (michelazzo) [reply pruatcher.py]: after all, looks like someone knows some python here, doesn't it?

File: pruatcher.py (hunk 3 above)
Comment (michelazzo) [reply pruatcher.py]: after all, looks like someone knows some python here, doesn't it?

**Comment by linehammer** [comment]
try it with....[list comprehension](http://net-informations.com/python/ds/comp.htm)

**Pull request merged**